injectproxy/routes.go

```diff
@@ -405,3 +420,27 @@ func matchersToString(ms ...*labels.Matcher) string {
 	}
 	return fmt.Sprintf("{%v}", strings.Join(el, ","))
 }
+
+type QueryParseError struct {
+	msg string
+}
+
+func (e QueryParseError) Error() string {
+	return e.msg
+}
+
+func newQueryParseError(err error) QueryParseError {
+	return QueryParseError{msg: fmt.Sprintf("Error parsing query string %s", err.Error())}
```

Comment:
Perhaps we should quote here to clarify where exactly the query starts and ends
```suggestion
	return QueryParseError{msg: fmt.Sprintf("Error parsing query string %q", err.Error())}
```

Reply:
Error strings typically start with lower case letters so they can be wrapped:
```suggestion
	return QueryParseError{msg: fmt.Sprintf("error parsing query string %s", err.Error())}
```

Reply:
Done